From 03fd0ecd914ba35a07925894336d7b5f766c65c4 Mon Sep 17 00:00:00 2001
From: Stanislav Levin <slev@altlinux.org>
Date: Thu, 10 Nov 2022 11:50:00 +0300
Subject: [PATCH] ALT: tests: Wait up to 30sec for the server start

- polling every 0.1sec is unreasonable overhead
- server start can take more than 14sec on slow platforms
---
 bind/bin/tests/system/start.pl | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/bin/tests/system/start.pl b/bin/tests/system/start.pl
index 657d9880807..1084a84349a 100755
--- a/bin/tests/system/start.pl
+++ b/bin/tests/system/start.pl
@@ -203,12 +203,12 @@ sub start_server {
 	my $child = `$command`;
 	chomp($child);
 
-	# wait up to 14 seconds for the server to start and to write the
+	# wait up to 30 seconds for the server to start and to write the
 	# pid file otherwise kill this server and any others that have
 	# already been started
 	my $tries = 0;
 	while (!-s $pid_file) {
-		if (++$tries > 140) {
+		if (++$tries > 30) {
 			print "I:$test:Couldn't start server $command (pid=$child)\n";
 			print "I:$test:failed\n";
 			kill "ABRT", $child if ("$child" ne "");
@@ -216,7 +216,7 @@ sub start_server {
 			system "$PERL $topdir/stop.pl $test";
 			exit 1;
 		}
-		sleep 0.1;
+		sleep 1;
 	}
 
 	# go back to the top level directory
-- 
2.33.5

